packages/tss/src/tss.ts

```diff
@@ -0,0 +1,677 @@
+      return ec.keyFromPublic({ x: p.x.toString(16, 64), y: p.y.toString(16, 64) }).getPublic();
+    });
+
+    const userDec = tssShareBNs[0];
+
+    const accountIndex = opts?.accountIndex || 0;
+    const coefficient = opts?.coefficient || new BN(1);
+    if (type === "direct") {
+      const tssSharePub = ec.g.mul(userDec);
+      const tssCommitA0 = tssCommits[0];
+      const tssCommitA1 = tssCommits[1];
+      const _tssSharePub = tssCommitA0.add(tssCommitA1.mul(new BN(tssIndex)));
+      if (tssSharePub.eq(_tssSharePub)) {
+        const adjustedShare = this.adjustTssShare(userDec, accountIndex, coefficient);
+        return { tssIndex, tssShare: adjustedShare };
+      }
+      throw new Error("user decryption does not match tss commitments...");
+    }
+
+    // if type === "hierarchical"
+    const serverDecs = tssShareBNs.slice(1); // 5 elems
+    const serverIndexes = new Array(serverDecs.length).fill(null).map((_, i) => i + 1);
+
+    const threshold = opts?.threshold || Math.ceil(serverDecs.length / 2);
+    const combis = kCombinations(serverDecs.length, threshold);
+    for (let i = 0; i < combis.length; i++) {
+      const combi = combis[i];
+      const selectedServerDecs = serverDecs.filter((_, j) => combi.indexOf(j) > -1);
+      if (selectedServerDecs.includes(null)) continue;
+
+      const selectedServerIndexes = serverIndexes.filter((_, j) => combi.indexOf(j) > -1);
+      const serverLagrangeCoeffs = selectedServerIndexes.map((x) => getLagrangeCoeffs(ec, selectedServerIndexes, x));
+      const serverInterpolated = dotProduct(serverLagrangeCoeffs, selectedServerDecs, ec.n);
+      const lagrangeCoeffs = [getLagrangeCoeffs(ec, [1, 99], 1), getLagrangeCoeffs(ec, [1, 99], 99)];
+      const tssShare = dotProduct(lagrangeCoeffs, [serverInterpolated, userDec], ec.n);
+      const tssSharePub = ec.g.mul(tssShare);
+      const tssCommitA0 = tssCommits[0];
+      const tssCommitA1 = tssCommits[1];
+      let _tssSharePub = tssCommitA0;
+      for (let j = 0; j < tssIndex; j++) {
+        _tssSharePub = _tssSharePub.add(tssCommitA1);
+      }
+      if (tssSharePub.eq(_tssSharePub)) {
+        const adjustedShare = this.adjustTssShare(tssShare, accountIndex, coefficient);
+        return { tssIndex, tssShare: adjustedShare };
+      }
+    }
+    throw new Error("could not find any combination of server decryptions that match tss commitments...");
+  }
+
+  getTSSCommits(): Point[] {
+    if (!this.privKey) throw CoreError.default("tss pub cannot be returned until you've reconstructed tkey");
+    if (!this.metadata) throw CoreError.metadataUndefined();
+    const tssPolyCommits = this.metadata.tssPolyCommits[this.tssTag];
+    if (!tssPolyCommits) throw CoreError.default(`tss poly commits not found for tssTag ${this.tssTag}`);
+    if (tssPolyCommits.length === 0) throw CoreError.default("tss poly commits is empty");
+    return tssPolyCommits;
+  }
+
+  getTSSPub(accountIndex?: number): Point {
```

Comment:
I think this should be fine to keep (the account index)..
what do you think?

Reply:
but to standardize the tssShare computation of the index using function, tssPublicKey probably should use similar method

Reply:
See the new `getTSSShare`. I decided to resolve the situation by allowing to provide a share coefficient. This way we don't need separate helper functions.